app/operations/aces/initiate_mec_check.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+
+module Aces
+  # Take a payload and find people. For each person, initiate a MecCheck
+  class InitiateMecCheck
+    send(:include, Dry::Monads[:result, :do])
+
+    # @param [String] payload received from Enroll
+    # @return [Dry::Result]
+    def call(payload)
+      json = JSON.parse(payload)
+      checks = yield get_person_check(json) if json["person"]
+      checks = yield get_people_checks(json) if json["people"]
+      results = yield save_check(checks, json)
+      publish_to_enroll(results)
+    end
+
+    protected
+
+    def get_person_check(json)
+      results = {}
+      person_hash = {}
+      person_hash["person"] = { person: json["person"] }
+      mc_response = mec_check(person_hash)
+      return Failure(mc_response.failure) if mc_response.failure?
+
+      results[json["person"]["hbx_id"]] = mc_response.value!
+      Success(results)
+    end
+
+    def get_people_checks(json)
+      results = {}
+      people = json["people"]
+      people.each do |person|
+        person_hash = { "person" => {} }
+        person_hash["person"]["person"] = person
+        mc_response = mec_check(person_hash)
+        return Failure(mc_response.failure) if mc_response.failure?
+
+        results[person["hbx_id"]] = mc_response.value!
+      end
+      Success(results)
+    end
+
+    def mec_check(person)
+      result = call_mec_check(person)
+      puts result
+      return Failure(result.failure) if result.failure?
+
+      response = JSON.parse(result.value!.to_json)
+      xml = Nokogiri::XML(response["body"])
+      response_description = xml.xpath("//xmlns:ResponseDescription", "xmlns" => "http://gov.hhs.cms.hix.dsh.ee.nonesi_mec.ext")
```

Comment:
May want to add something here as a fallback if the response doesn't have the tag. if so, it will break on the .text